uplink/src/collector/telemetry.rs

```diff
@@ -0,0 +1,534 @@
+    }
+
+    fn anomalies(&self) -> Option<(String, usize)> {
+        self.anomalies()
+    }
+}
+
+#[derive(Debug, Default, Serialize, Clone)]
+struct Processor {
+    sequence: u32,
+    timestamp: u64,
+    name: String,
+    frequency: u64,
+    usage: f32,
+}
+
+impl Processor {
+    fn init(name: String) -> Self {
+        Processor { name, ..Default::default() }
+    }
+
+    fn update(&mut self, proc: &sysinfo::Processor, timestamp: u64, sequence: u32) {
+        self.frequency = proc.frequency();
+        self.usage = proc.cpu_usage();
+        self.timestamp = timestamp;
+        self.sequence = sequence;
+    }
+}
+
+impl Point for Processor {
+    fn sequence(&self) -> u32 {
+        self.sequence
+    }
+
+    fn timestamp(&self) -> u64 {
+        self.timestamp
+    }
+}
+
+impl Package for Buffer<Processor> {
+    fn topic(&self) -> Arc<String> {
+        self.topic.clone()
+    }
+
+    fn serialize(&self) -> Vec<u8> {
+        serde_json::to_vec(&self.buffer).unwrap()
+    }
+
+    fn anomalies(&self) -> Option<(String, usize)> {
+        self.anomalies()
+    }
+}
+
+#[derive(Debug, Default, Serialize, Clone)]
+struct DiskUsage {
+    total_written_bytes: u64,
+    written_bytes: u64,
+    total_read_bytes: u64,
+    read_bytes: u64,
+}
+
+#[derive(Debug, Default, Serialize, Clone)]
+struct Process {
+    sequence: u32,
+    timestamp: u64,
+    pid: i32,
+    name: String,
+    cpu_usage: f32,
+    mem_usage: u64,
+    #[serde(flatten)]
+    disk_usage: DiskUsage,
+    start_time: u64,
+}
+
+impl Process {
+    fn init(pid: i32, name: String, start_time: u64) -> Self {
+        Process { pid, name, start_time, ..Default::default() }
+    }
+
+    fn update(&mut self, proc: &sysinfo::Process, timestamp: u64, sequence: u32) {
+        let sysinfo::DiskUsage { total_written_bytes, written_bytes, total_read_bytes, read_bytes } =
+            proc.disk_usage();
+        self.disk_usage =
+            DiskUsage { total_written_bytes, written_bytes, total_read_bytes, read_bytes };
+        self.cpu_usage = proc.cpu_usage();
+        self.mem_usage = proc.memory();
+        self.timestamp = timestamp;
+        self.sequence = sequence;
+    }
+}
+
+impl Point for Process {
+    fn sequence(&self) -> u32 {
+        self.sequence
+    }
+
+    fn timestamp(&self) -> u64 {
+        self.timestamp
+    }
+}
+
+impl Package for Buffer<Process> {
+    fn topic(&self) -> Arc<String> {
+        self.topic.clone()
+    }
+
+    fn serialize(&self) -> Vec<u8> {
+        serde_json::to_vec(&self.buffer).unwrap()
+    }
+
+    fn anomalies(&self) -> Option<(String, usize)> {
+        self.anomalies()
+    }
+}
+
+type Sequence = u32;
+
+/// Holds current sequence number values meant for different data streams
+#[derive(Default)]
+struct Sequences {
+    system: Sequence,
+    processors: Sequence,
+    processes: Sequence,
+    disks: Sequence,
+    networks: Sequence,
+}
+
+/// Stream handles for the various data streams
+struct Streams {
```

Comment:
This should be renamed to `SystemStats` or something else. Definitely not `Streams`

Reply:
How about `StatStreams`?